src/main/java/com/ibm/eventstreams/connect/jdbcsink/database/writer/JDBCWriter.java

```diff
@@ -50,107 +51,205 @@ private boolean doesTableExist(Connection connection, String tableName) throws S
         return table.next();
     }
 
-    // TODO: verify will work for all database flavors
-    private String getDatabaseFieldType(Schema.Type type) {
-        switch (type) {
+    private boolean isPostgreSQL(Connection connection) throws SQLException {
+        return connection.getMetaData().getDatabaseProductName().toLowerCase().contains("postgresql");
+    }
+
+    private boolean isDB2(Connection connection) throws SQLException {
+        return connection.getMetaData().getDatabaseProductName().toLowerCase().contains("db2");
+    }
+
+    private boolean isMySQL(Connection connection) throws SQLException {
+        return connection.getMetaData().getDatabaseProductName().toLowerCase().contains("mysql");
+    }
+
+    private String getPostgreSQLFieldType(Schema.Type fieldType) {
+        switch (fieldType) {
+            case INT8:
+            case INT16:
+            case INT32:
+                return "INTEGER";
+            case INT64:
+                return "BIGINT";
+            case FLOAT32:
+                return "REAL";
+            case FLOAT64:
+                return "DOUBLE PRECISION";
+            case BOOLEAN:
+                return "BOOLEAN";
+            case STRING:
+                return "VARCHAR(255)";
+            default:
+                return "VARCHAR(255)";
+        }
+    }
+
+    private String getDB2FieldType(Schema.Type fieldType) {
+        switch (fieldType) {
             case INT8:
             case INT16:
             case INT32:
                 return "INTEGER";
             case INT64:
                 return "BIGINT";
             case FLOAT32:
+                return "REAL";
             case FLOAT64:
+                return "DOUBLE";
+            case BOOLEAN:
+                return "BOOLEAN";
+            case STRING:
+                return "VARCHAR(255)";
+            default:
+                return "VARCHAR(255)";
+        }
+    }
+
+    private String getMySQLFieldType(Schema.Type fieldType) {
+        switch (fieldType) {
+            case INT8:
+            case INT16:
+            case INT32:
+                return "INT";
+            case INT64:
+                return "BIGINT";
+            case FLOAT32:
                 return "FLOAT";
+            case FLOAT64:
+                return "DOUBLE";
             case BOOLEAN:
                 return "BOOLEAN";
+            case STRING:
+                return "VARCHAR(255)";
             default:
                 return "VARCHAR(255)";
         }
     }
 
+    private String getIdColumnDefinition() {
```

Comment:
Would it be reasonable to have an abstract command builder separate from the writer?

Reply:
Yes, this would be a great addition to have a separate command builder.